stripe/src/main/java/com/stripe/android/net/CardParser.java

```diff
@@ -0,0 +1,74 @@
+package com.stripe.android.net;
+
+import android.support.annotation.NonNull;
+
+import com.stripe.android.model.Card;
+import com.stripe.android.util.StripeJsonUtils;
+import com.stripe.android.util.StripeTextUtils;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * A helper class for parsing {@link Card} objects returned from the server.
+ */
+public class CardParser {
+
+    static final String FIELD_ADDRESS_CITY = "address_city";
```

Comment:
There exists constructors for JSONObject that *can* do this correctly for primitives, but my trust in that particular constructor is limited by how many times I've seen it break for stupid reasons. GSON has a very robust way of doing this, but we're flying solo here.

There is also the matter of the `Card` object needing to be recursively parsed in the `Token` object, and the native JSON libraries are finicky about that.